src/Compilers/CSharp/Portable/Binder/Binder_Query.cs

```diff
@@ -928,6 +928,8 @@ protected BoundCall MakeQueryInvocation(CSharpSyntaxNode node, BoundExpression r
             Debug.Assert(receiver.Type is object || ultimateReceiver.Type is null);
             if ((object?)ultimateReceiver.Type == null)
             {
+                Debug.Assert(ultimateReceiver.Kind != BoundKind.MethodGroup || ultimateReceiver.HasAnyErrors);
```

Comment:
:memo: See previous discussion on this change [https://github.com/dotnet/roslyn/pull/73239#discussion_r1586625386](https://github.com/dotnet/roslyn/pull/73239#discussion_r1586625386)